File: packages/native-modules/android/src/main/java/expo/modules/nativeuiorgnativemodules/tabs/NativeUiOrgTabsModule.kt
package expo.modules.nativeuiorgnativemodules.tabs

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeUiOrgTabsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NativeUiOrgTabs")

    // Define the native view component
    View(NativeUiOrgTabsView::class) {
      // Props
      Prop("tabs") { view: NativeUiOrgTabsView, items: List<Map<String, Any>> ->
        view.setTabs(items)
      }
      
      Prop("selectedValue") { view: NativeUiOrgTabsView, value: String ->
        view.setSelectedValue(value)
      }
      
      // Events
      Events("onValueChange")
    }
  }
}

File: packages/native-modules/ios/tabs/NativeUiOrgTabsModule.swift
import ExpoModulesCore

public class NativeUiOrgTabsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NativeUiOrgTabs")

    // Define the native view component
    View(NativeUiOrgTabsView.self) {
      // Props
      Prop("tabs") { (view: NativeUiOrgTabsView, items: [[String: Any]]) in
        view.setTabs(items)
      }
      
      Prop("selectedValue") { (view: NativeUiOrgTabsView, value: String) in
        view.setSelectedValue(value)
      }
      
      // Customization props
      Prop("variant") { (view: NativeUiOrgTabsView, variant: String) in
        view.setVariant(variant)
      }
      
      Prop("alignment") { (view: NativeUiOrgTabsView, alignment: String) in
        view.setAlignment(alignment)
      }
      
      Prop("chevronIcon") { (view: NativeUiOrgTabsView, icon: String) in
        view.setChevronIcon(icon)
      }
      
      Prop("showChevron") { (view: NativeUiOrgTabsView, show: Bool) in
        view.setShowChevron(show)
      }
      
      Prop("backgroundColor") { (view: NativeUiOrgTabsView, color: String?) in
        view.setBackgroundColor(color)
      }
      
      Prop("textColor") { (view: NativeUiOrgTabsView, color: String?) in
        view.setTextColor(color)
      }
      
      Prop("borderColor") { (view: NativeUiOrgTabsView, color: String?) in
        view.setBorderColor(color)
      }
      
      Prop("cornerRadius") { (view: NativeUiOrgTabsView, radius: Double) in
        view.setCornerRadius(radius)
      }
      
      // Events
      Events("onValueChange")
    }
  }
}

